lib/wallet/txdb.js

```diff
@@ -79,6 +79,62 @@ class TXDB {
     this.wallet.emit(event, data, details);
   }
 
+  /**
+   * Emit events for each type of covenant.
+   * @private
+   * @param {TXRecord} wtx
+   * @param {Details} details
+   */
+
+  async emitCovenants(tx, details, view, height) {
+    const network = this.wdb.network;
+    if (height < 0)
+      height = null;
+
+    const {outputs} = tx;
+    for (const output of outputs) {
+      const {covenant} = output;
+      if (!covenant.isName())
+        continue;
+
+      const nameHash = covenant.get(0);
+      const ns = await view.getNameState(this, nameHash);
+
+      if (EMPTY.equals(ns.name) || !ns.name) {
+        switch (covenant.type) {
+          case types.CLAIM:
+          case types.OPEN:
+          case types.BID:
+          case types.FINALIZE: {
+            ns.name = covenant.get(2);
+            break;
+          }
+          case types.REVEAL:
+          case types.REDEEM:
+          case types.REGISTER:
+          case types.UPDATE:
+          case types.RENEW:
+          case types.TRANSFER:
+          case types.REVOKE: {
+            const nameHash = covenant.get(0);
```

Comment:
Delete `nameHash` variable assignment here, there is another above the `switch` statement.